### PR TITLE
Enhance Subscription Filter with Lambda Function Separation

### PR DESCRIPTION
**Pull request by bardabun**
## Changes

This PR enhances the subscription filter feature by introducing two distinct Lambda functions: one for handling EventBridge events and another for CloudFormation events. This separation allows for more flexibility and direct updates to service and custom log parameters within the stack, improving maintainability and ease of use.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,4 +1,4 @@
-module main
+module firehose-logs
```
Comment (yotamloe) [inline go.mod:1]: This `go.mod` file is redundant here. Please add `go.mod` to `cloudformation` and `eventbridge-lambda` and name the modules according to this pattern `github.com/logzio/<<repo-name>>/<<path-to-module>>` see this example: https://github.com/logzio/prometheus-alerts-migrator/blob/master/go.mod#L1

File: common/utils.go [hunk 2]
```diff
@@ -8,7 +8,7 @@ import (
 const (
 	envServices                  = "SERVICES"
```
Comment (yotamloe) [inline common/utils.go:9]: Where are these private consts in use? I didn't find any reference of them in other modules.

**Review by yotamloe — CHANGES_REQUESTED** [review]
Amazing job! see my comments

File: go.mod (hunk 1 above)
Comment (bardabun) [reply go.mod:1]: I've updated the module path in the `go.mod` file as suggested. However, I'd like to discuss the recommendation to add separate `go.mod` files for each Lambda function. Given that our Lambda functions share a significant amount of code and dependencies, I believe maintaining a single `go.mod` at the project root simplifies dependency management, reduces maintenance overhead, and enhances project cohesion. This approach seems more manageable for our current project structure, ensuring consistent dependency versions across all components and streamlining our build and deployment processes. 
Could we explore the benefits of separate `go.mod` files for our specific use case, considering these points?

Comment (yotamloe) [reply go.mod:1]: If you think we need to manage the dependencies of all of the function included (now and in the future) together, I Can accept it. Its fine if they share significant amount of code and dependencies. Anyway rename the module to the pattern I mentioned

**Review by ralongit — APPROVED** [review]

**Pull request merged**